src/logics/Logic.cc

```diff
@@ -73,138 +73,27 @@ Logic::Logic() :
       distinctClassCount(0)
     , sort_store(id_store)
     , term_store(sym_store)
-    , sym_TRUE(SymRef_Undef)
-    , sym_FALSE(SymRef_Undef)
-    , sym_ANON(SymRef_Undef)
-    , sym_AND(SymRef_Undef)
-    , sym_OR(SymRef_Undef)
-    , sym_XOR(SymRef_Undef)
-    , sym_NOT(SymRef_Undef)
-    , sym_UF_NOT(SymRef_Undef)
-    , sym_EQ(SymRef_Undef)
-    , sym_IMPLIES(SymRef_Undef)
-    , sym_DISTINCT(SymRef_Undef)
-    , sym_ITE(SymRef_Undef)
-    , sort_BOOL(SRef_Undef)
-    , term_TRUE(PTRef_Undef)
-    , term_FALSE(PTRef_Undef)
+    , sort_BOOL(sort_store.newSort(id_store.newIdentifier(s_sort_bool), {}))
+    , term_TRUE(mkConst(getSort_bool(), tk_true))
+    , term_FALSE(mkConst(getSort_bool(), tk_false))
+    , sym_TRUE(getSymRef(term_TRUE))
+    , sym_FALSE(getSymRef(term_FALSE))
+    , sym_ANON(sym_store.newSymb(tk_anon, {}))
+    , sym_AND(declareFun_Commutative_NoScoping_LeftAssoc(tk_and, sort_BOOL, {sort_BOOL, sort_BOOL}))
+    , sym_OR(declareFun_Commutative_NoScoping_LeftAssoc(tk_or, sort_BOOL, {sort_BOOL, sort_BOOL}))
+    , sym_XOR(declareFun_Commutative_NoScoping_LeftAssoc(tk_xor, sort_BOOL, {sort_BOOL, sort_BOOL}))
+    , sym_NOT(declareFun_NoScoping(tk_not, sort_BOOL, {sort_BOOL}))
+    , sym_UF_NOT(declareFun_NoScoping(tk_uf_not, sort_BOOL, {sort_BOOL}))
+    , sym_EQ(declareFun_Commutative_NoScoping_Chainable(tk_equals, sort_BOOL, {sort_BOOL, sort_BOOL}))
+    , sym_IMPLIES(declareFun_NoScoping_RightAssoc(tk_implies, sort_BOOL, {sort_BOOL, sort_BOOL}))
+    , sym_DISTINCT(declareFun_Commutative_NoScoping_Pairwise(tk_distinct, sort_BOOL, {sort_BOOL, sort_BOOL}))
+    , sym_ITE(declareFun_NoScoping_LeftAssoc(tk_ite, sort_BOOL, {sort_BOOL, sort_BOOL}))
```

Comment:
Here `ITE` symbol gets the `left associativity` flag, but in `declareSortAndCreateFunctions` it does not. Why?

Reply:
It's because the this is our special left-associative boolean Ite that only takes two arguments, instead of the standard one you're thinking of that takes three arguments 😄 

It's a bug, hopefully now fixed.